src/ext/java/org/opentripplanner/ext/siri/SiriTripPatternCache.java

```diff
@@ -157,7 +157,7 @@ private String generateUniqueTripPatternCode(TripPattern tripPattern) {
             counter++;
         }
         // OBA library uses underscore as separator, we're moving toward colon.
-        String code = String.format("%s:%s:%s:rt#%d", routeId.getFeedId(), routeId.getId(), direction, counter);
+        String code = String.format("%s:%s:rt#%d", routeId.getId(), direction, counter);
```

Comment:
This is dropping the feed scope, but I guess realtime sources are always matched to one specific feed so this is not needed, and in any case if we did want it we would use FeedScopedId instead of prepending another string.

Reply:
I agree that once the reference to the TransitLayer has been handed off, it should not change. As you suggested, I have now changed the TransitLayerUpdater to make a shallow copy of of the TransitLayer before updating it. As none of the objects that are referenced at the top level of the TransitLayer (except tripPatternsForDate, which is also copied in the TransitLayer constructor), a shallow copy is enough to ensure that the original object is not changed.